Source/Core/Core/NetPlayCommon.cpp

```diff
@@ -187,6 +217,37 @@ bool DecompressPacketIntoFile(sf::Packet& packet, const std::string& file_path)
   return true;
 }
 
+static bool DecompressPacketIntoFolderInternal(sf::Packet& packet, const std::string& folder_path)
+{
+  if (!File::CreateFullPath(folder_path + "/"))
+    return false;
+
+  sf::Uint64 size;
+  packet >> size;
+  for (size_t i = 0; i < size; ++i)
+  {
+    std::string name;
+    packet >> name;
+    bool is_folder;
+    packet >> is_folder;
+    std::string path = fmt::format("{}/{}", folder_path, name);
```

Comment:
Path traversal vulnerability.

(I wish there was a convenient API for escaping/filtering dangerous characters... SQL got this down a pretty long time ago with prepared statements, but for paths there doesn't really seem to be anything in major OSes/libraries.)

Reply:
Checking for '/' and '\' in `name` is probably good enough here?

Reply:
Oh no wait, it can't be all dots either... bleh.